src/Aspire.Hosting/DistributedApplicationOptions.cs

```diff
@@ -39,7 +41,11 @@ public DistributedApplicationOptions()
 
     internal Assembly? Assembly => _assembly.Value;
 
-    internal string? ProjectDirectory => _projectDirectory.Value;
+    internal string? ProjectDirectory
+    {
+        get => _projectDirectory ?? _projectDirectoryLazy.Value;
```

Comment:
```suggestion
        get => _projectDirectory ??= _projectDirectoryLazy.Value;
```

Might as well set this after you evaluate it, right?

Another thought is to remove the field altogether. I don't see it being used in tests. Am I missing something?

Reply:
Yea the property is being used in tests.

Reply:
It's confusing to mark the field with the comment

Reply:
What do you mean? This is only a field so we can set it.

Reply:
Typically when someone puts a comment like:

```C#
// for testing
internal Foo() {}
```

It means Foo is getting called/used by tests.

A whole other option is to just use a single field, and when it is `null` during `get` generate the value.